opal/include/opal/sys/gcc_builtin/atomic.h

```diff
@@ -193,26 +193,30 @@ static inline intptr_t opal_atomic_swap_ptr(opal_atomic_intptr_t *addr, intptr_t
 
 static inline void opal_atomic_lock_init(opal_atomic_lock_t *lock, int32_t value)
 {
-    lock = value;
+    if (OPAL_ATOMIC_LOCK_UNLOCKED == value) {
+        __atomic_clear(lock, __ATOMIC_RELAXED);
+    } else {
+        *lock = true;
+    }
 }
 
 static inline int opal_atomic_trylock(opal_atomic_lock_t *lock)
 {
-    int ret = __atomic_exchange_n(&lock, OPAL_ATOMIC_LOCK_LOCKED,
-                                  __ATOMIC_ACQUIRE | __ATOMIC_HLE_ACQUIRE);
-    if (OPAL_ATOMIC_LOCK_LOCKED == ret) {
+    bool ret = __atomic_test_and_set(&lock, OPAL_ATOMIC_LOCK_LOCKED,
```

Comment:
This all should be in its own commit with its own justification.  This code was used in production prior to the 5.0 series, so it feels like (without justification in a commit message or comment) this is changing things to change things.